core/src/test/java/com/amplifyframework/analytics/PropertyNamesTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class PropertyNamesTest {
+
+    @Test
+    @SuppressWarnings("MagicNumber")
+    public void of() {
+        // Arrange: Use the static factory method to create a PropertyNames instance
+        Set<String> names = new HashSet<>(Arrays.asList("Height", "Weight", "StarSign"));
+
+        PropertyNames propertyNames;
+        propertyNames = PropertyNames.of(names.toArray(new String[names.size()]));
+
+        // Assert: Iterate through the items and remove each name from the set, then check all items
+        // were removed after iteration
+        assertEquals(3, (long) propertyNames.size());
```

Comment:
(By the time you get this returning a primitive, I think the case will go away, and the `assertEquals(int, int)` would be automatically selected, instead of `assertEquals(long, long)`, as it is now (or `assertEquals(Object, Object)`, as it might have been without the primitive casting.)